$OpenBSD: patch-coders_tiff_c,v 1.1 2012/04/27 08:09:56 jasper Exp $

Security fix for CVE-2012-1798, Copying of invalid memory when reading TIFF EXIF IFD.

--- coders/tiff.c.orig	Fri Apr 27 09:42:38 2012
+++ coders/tiff.c	Fri Apr 27 09:43:24 2012
@@ -596,7 +596,7 @@ static void TIFFGetEXIFProperties(TIFF *tiff,Image *im
           *ascii;
 
         if (TIFFGetField(tiff,exif_info[i].tag,&ascii) != 0)
-          (void) CopyMagickMemory(value,ascii,MaxTextExtent);
+          (void) CopyMagickString(value,ascii,MaxTextExtent);
         break;
       }
       case TIFF_SHORT:
